src/math/CurveFitting.cc

```diff
@@ -495,215 +488,115 @@ namespace pfs{ namespace drp{ namespace stella{ namespace math{
       if (P_D_A2_Covar->getShape()[0] != nCoeffs){
         string message("pfs::drp::stella::math::CurveFitting::PolyFit: ERROR: P_D_A2_Covar->getShape()[0](=");
         message += to_string(P_D_A2_Covar->getShape()[0]) + ") != nCoeffs(=" + to_string(nCoeffs) + ")";
-        cout << message << endl;
         throw LSST_EXCEPT(pexExcept::Exception, message.c_str());
       }
       if (P_D_A2_Covar->getShape()[1] != nCoeffs){
         string message("pfs::drp::stella::math::CurveFitting::PolyFit: ERROR: P_D_A2_Covar->getShape()[1](=");
         message += to_string(P_D_A2_Covar->getShape()[1]) + ") != nCoeffs(=" + to_string(nCoeffs) + ")";
-        cout << message << endl;
         throw LSST_EXCEPT(pexExcept::Exception, message.c_str());
       }
-      #ifdef __DEBUG_POLYFIT__
-        cout << "pfs::drp::stella::math::CurveFitting::PolyFit: KeyWord_Set(COVAR): *P_D_A2_Covar set to " << (*P_D_A2_Covar) << endl;
-      #endif
+      LOGLS_DEBUG(_log, "KeyWord_Set(COVAR): *P_D_A2_Covar set to " << (*P_D_A2_Covar));
     }
+    P_D_A2_Covar->deep() = 0.;
```

Comment:
Again, why is it only necessary to initialize this now? Please explain the motivation. If it's a bug fix, can we have a unit test?